### PR TITLE
decompile EntityCastleBridge

### PR DESCRIPTION
**Pull request by Pyromuffin**
this one was trickier and larger than my previous contributions. I'm not sure what the best way to rename the symbols are, so I expect to receive some comments on that.

### CODE REVIEW

File: src/st/no3/3C4EC.c [hunk 1]
```diff
@@ -326,7 +326,230 @@ void EntityUnkId52(Entity* self) {
     }
 }
 
-INCLUDE_ASM("st/no3/nonmatchings/3C4EC", EntityCastleBridge);
+// func_psp_0923F328
+void EntityCastleBridge(Entity* self) {
+
+#define priorities D_801819A0
```
Comment (sozud) [inline src/st/no3/3C4EC.c:332]: Could we rename the actual variables? Add your new names to https://github.com/Xeeynamo/sotn-decomp/blob/master/config/symbols.us.stno3.txt

File: src/st/no3/3C4EC.c [hunk 2]
```diff
@@ -326,7 +326,230 @@ void EntityUnkId52(Entity* self) {
...
+                case 0:
+                    if (player->velocityY < 0) {
+                        self->step_s++;
+                    }
+                    break;
+                case 1:
+                    if (player->velocityY >= 0 &&
+                        (xOffset < player->posY.i.hi + 30)) {
+                        player->posY.i.hi = xOffset - 30;
+                        g_api.PlaySfx(0x64B);
```
Comment (sozud) [inline src/st/no3/3C4EC.c:432]: SFX_STOMP_A I think? https://github.com/Xeeynamo/sotn-decomp/blob/3ae3f7576744e3d1889e95260a6f3acb24891da7/include/sfx.h#L383C5-L383C21

File: src/st/no3/3C4EC.c [hunk 3]
```diff
@@ -326,7 +326,230 @@ void EntityUnkId52(Entity* self) {
...
+            prim->y0 = (s16)rotatedVector->vy + 168;
+
+            vector->vx = *positionsPtr++;
+            vector->vy = *positionsPtr++;
+            ApplyRotMatrix(vector, rotatedVector);
+
+            prim->x2 = (s16)rotatedVector->vx + xOffset;
+            prim->y2 = (s16)rotatedVector->vy + 168;
+        } else {
+            // size |= 1;
```
Comment (sozud) [inline src/st/no3/3C4EC.c:479]: Is this here intentionally?

File: config/symbols.us.stno3.txt [hunk 4]
```diff
@@ -15,6 +15,10 @@ g_eBreakableExplosionTypes = 0x80180E78;
 g_eBreakableanimSets = 0x80180E80;
 g_eBreakableDrawModes = 0x80180E90;
 g_TrapDoorFlag = 0x8018123C;
+EntityCastleBridgeUVs = 0x80181940;
```
Comment (sozud) [inline config/symbols.us.stno3.txt:18]: Would you mind adding `g_` to match our style? https://github.com/Xeeynamo/sotn-decomp/blob/master/docs/STYLE.md#naming

Other than that looks good!

**Review by sozud — APPROVED** [review]

File: config/symbols.us.stno3.txt (hunk 4 above)
Comment (Pyromuffin) [reply config/symbols.us.stno3.txt]: yeah, it was a bit unclear how to name these, whether they are truly global. Thanks for the review!

**Pull request merged**